app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -308,7 +309,14 @@ public void onClick(DialogInterface dialog, int which) {
     @Override
     protected void onResume() {
         super.onResume();
+
         settings = GoIVSettings.getInstance(MainActivity.this);
+
+        if (isPokeFlyRunning()) {
+            launchButton.setText(R.string.main_stop);
+        } else {
+            launchButton.setText(R.string.main_start);
+        }
```

Comment:
Also, the existing logic for starting the service only works correctly if the label is correct (which seems... weird to me). @DJCrapsody in your first scenario, is it possible the label hadn't been updated? Because the logic to start the service from the button is unchanged.